src/transaction/buffer.rs

```diff
@@ -158,7 +159,12 @@ impl Buffer {
             .into_iter()
             .map(|(k, v)| KvPair::new(k, v))
             .collect::<Vec<_>>();
-        res.sort_by_cached_key(|x| x.key().clone());
+
+        if reverse {
```

Comment:
Would you mind adding a TODO here? Since we can use BTreeMap to eliminate this sort.

Reply:
OK, I will add a TODO note here